### ISSUE TITLE
[Bug]: Crash while Creating UI

### ISSUE THREAD

**Issue by Dteyn** [originalnicodr/CinematicUnityExplorer#70]
### Are you on the latest version of CinematicUnityExplorer?

- [X] Yes, I'm on the latest version of CinematicUnityExplorer.

### Which release are you using?

MelonLoader IL2CPP

### Which game did this occur on?

Game Name: AfterTheFall
Game Developer: Vertigo Games
Unity Version: 2021.3.8f1
App Version: 1.0
Game Version: 1.9.41947

### Describe the issue.

Hi! I recently discovered CUE and fell in love with the idea of using it for a video project I'm working on. It seems to offer everything I could have hoped for in a freecam and would love to use it. However, it seems to crash when loading with the game I'm trying to use it with.

The game I'm using it with has been working well with UnityExplorer v4.9.0 with and doesn't seem to have any issues with that. Everything seems to work normally with my usual UE v4.9.0 setup (using ML v0.5.7). 

The version of CUE I am using is the ML 0.5 IL2CPP version, `CinematicUnityExplorer.MelonLoader.IL2CPP.zip`.

However, with CUE, it seems to crash during startup. I'm not sure if this is the specific error causing the crash, but I see this close to the end of the log file:

```
[21:49:36.748] [CinematicUnityExplorer] [UniverseLib] Initialized Legacy Input support
[21:49:36.791] [CinematicUnityExplorer] [UniverseLib] 	 IL2CPP method has no corresponding pointer, aborting patch of UnityEngine.AssetBundle.UnloadAllAssetBundles
```

The game actually seems to be loading fine for about 1-2s until the crash happens. Also, just before the crash, there are two lines which are shown in the MelonLoader logging window which aren't included in the `MelonLoader\Latest.log` file. They only appear briefly before the window disappears, so I did a quick screen capture:

![image](https://github.com/originalnicodr/CinematicUnityExplorer/assets/133910763/9595a659-8a0d-4528-a03a-5fcc4bb72afe)

The last two lines are:

```
mono_class_from_mono_type: implement me 0x00

* Assertion: should not be reached at ..\mono\metadata\class.c:6545
```

I tried what you suggested in the "Common issues and solutions" section, by changing `Startup_Delay_Time` to 20.0, and changing `Disable_EventSystem_Override` to `true`, but unfortunately it doesn't seem to make any difference. When I check the game's `Player.log` file, I don't see anything unusual there, but I'm happy to share it if you think it would help.

A couple of other things I've tried:

- Using MelonLoader v0.6.2 - it also has the same issue, however I prefer to use ML v0.5.7 due to some other tools I'd like to use in conjunction with CUE once I get it loaded/working (note: I'm not trying to use anything else at the moment while I get CUE set up)
- Tried copying UniverseLib v1.5.1 from my working UE v4.9.0 installation - no change, and not sure if this would have worked as just a random guess since UE v4.9.0 is working for me
- Tried the CoreCLR and .net6 preview versions (both crashed when launching - I'm sure these are not the correct versions for my setup, as the game uses .net472 but I wanted to try anyway)
- Re-installed everything and checked files to make sure it's all installed correctly

I'm happy to try any other suggestions, or provide any other information you need.

Thanks for your hard work on this - I would love to be able to get this working for my project. Using UE's freecam has been a challenge for sure and all the work you've put into this is like a dream come true for what I'd like to do. Thanks again!!

### Relevant log output

```shell
[21:49:12.627] ------------------------------
[21:49:12.628] MelonLoader v0.5.7 Open-Beta
[21:49:12.628] OS: Windows 10
[21:49:12.628] Hash Code: 49999848531024954985654485098975250521005157544948501005510199495356
[21:49:12.629] ------------------------------
[21:49:12.629] Game Type: Il2Cpp
[21:49:12.629] Game Arch: x64
[21:49:12.630] ------------------------------
[21:49:12.630] Core::BasePath = F:\SteamLibrary\steamapps\common\After The Fall
[21:49:12.630] Game::BasePath = F:\SteamLibrary\steamapps\common\After The Fall
[21:49:12.631] Game::DataPath = F:\SteamLibrary\steamapps\common\After The Fall\AfterTheFall_Data
[21:49:12.631] Game::ApplicationPath = F:\SteamLibrary\steamapps\common\After The Fall\AfterTheFall.exe

[21:49:12.867] ------------------------------
[21:49:12.867] Game Name: AfterTheFall
[21:49:12.867] Game Developer: Vertigo Games
[21:49:12.870] Unity Version: 2021.3.8f1
[21:49:12.870] Game Version: 1.0
[21:49:12.870] ------------------------------

[21:49:13.091] Preferences Loaded!
[21:49:13.095] [Il2CppUnityTls] Patching mono_unity_get_unitytls_interface...
[21:49:13.096] [Il2CppUnityTls] Patching il2cpp_unity_install_unitytls_interface...
[21:49:13.245] [Il2CppUnityTls] Calling Il2CppInstallUnityTlsInterface...

[21:49:13.259] Loading Plugins from 'F:\SteamLibrary\steamapps\common\After The Fall\Plugins'...
[21:49:13.260] 0 Plugins loaded.

[21:49:14.291] [MelonStartScreen] Initializing...
[21:49:14.305] [MelonStartScreen] Using Start Screen Theme: "Default"
[21:49:15.232] Loading Il2CppAssemblyGenerator...
[21:49:15.286] [Il2CppAssemblyGenerator] Contacting RemoteAPI...
[21:49:15.578] [Il2CppAssemblyGenerator] Game Not Found on RemoteAPI Host (https://api.melonloader.com/api/v1/game/afterthefall)
[21:49:15.579] [Il2CppAssemblyGenerator] RemoteAPI.DumperVersion = null
[21:49:15.580] [Il2CppAssemblyGenerator] RemoteAPI.UnhollowerVersion = null
[21:49:15.581] [Il2CppAssemblyGenerator] RemoteAPI.ObfuscationRegex = null
[21:49:15.582] [Il2CppAssemblyGenerator] RemoteAPI.MappingURL = null
[21:49:15.582] [Il2CppAssemblyGenerator] RemoteAPI.MappingFileSHA512 = null
[21:49:15.584] [Il2CppAssemblyGenerator] Using Dumper Version: 2022.1.0-pre-release.8
[21:49:15.585] [Il2CppAssemblyGenerator] Using Il2CppAssemblyUnhollower Version = 0.4.18.0
[21:49:15.585] [Il2CppAssemblyGenerator] Using Unity Dependencies Version = 2021.3.8
[21:49:15.586] [Il2CppAssemblyGenerator] Using Deobfuscation Regex = null
[21:49:15.587] [Il2CppAssemblyGenerator] Cpp2IL is up to date.
[21:49:15.588] [Il2CppAssemblyGenerator] Il2CppAssemblyUnhollower is up to date.
[21:49:15.588] [Il2CppAssemblyGenerator] UnityDependencies is up to date.
[21:49:15.589] [Il2CppAssemblyGenerator] Checking GameAssembly...
[21:49:16.176] [Il2CppAssemblyGenerator] Assembly is up to date. No Generation Needed.


[21:49:16.203] Loading Mods from 'F:\SteamLibrary\steamapps\common\After The Fall\Mods'...
[21:49:16.204] ------------------------------
[21:49:16.226] Melon Assembly loaded: '.\Mods\CinematicUnityExplorer.ML.IL2CPP.dll'
[21:49:16.227] SHA256 Hash: '93637ded82b6288760e35d28f31a5da4518f91d6d670ce43fc7f59d6c443ffff'

[21:49:16.251] ------------------------------
[21:49:16.251] CinematicUnityExplorer v1.0.0
[21:49:16.252] by originalnicodr, Sinai, yukieiji
[21:49:16.253] Assembly: CinematicUnityExplorer.ML.IL2CPP.dll
[21:49:16.253] ------------------------------
[21:49:16.254] ------------------------------
[21:49:16.254] 1 Mod loaded.

[21:49:16.351] Support Module Loaded: F:\SteamLibrary\steamapps\common\After The Fall\MelonLoader\Dependencies\SupportModules\Il2Cpp.dll
[21:49:16.372] [CinematicUnityExplorer] CinematicUnityExplorer 1.0.0 initializing...
[21:49:16.388] [CinematicUnityExplorer] [UniverseLib] UniverseLib 1.5.5 initializing...
[21:49:16.617] [CinematicUnityExplorer] [UniverseLib] Finished UniverseLib initial setup.
[21:49:16.667] [CinematicUnityExplorer] Initialized UnityCrashPrevention for: Canvas.get_renderingDisplaySize
[21:49:25.811] [CinematicUnityExplorer] [UniverseLib] Loaded Unhollowed modules in 9.209001 seconds.
[21:49:25.949] [CinematicUnityExplorer] [UniverseLib] Setup deobfuscation cache in 0.137 seconds.
[21:49:36.748] [CinematicUnityExplorer] [UniverseLib] Initialized Legacy Input support
[21:49:36.791] [CinematicUnityExplorer] [UniverseLib] 	 IL2CPP method has no corresponding pointer, aborting patch of UnityEngine.AssetBundle.UnloadAllAssetBundles
[21:49:36.845] [CinematicUnityExplorer] [UniverseLib] Loaded modern bundle for Unity 2021.3.8f1
[21:49:36.869] [CinematicUnityExplorer] [UniverseLib] UniverseLib 1.5.5 initialized.
[21:49:36.870] [CinematicUnityExplorer] Creating UI...
mono_class_from_mono_type: implement me 0x00

* Assertion: should not be reached at ..\mono\metadata\class.c:6545
```

**Comment by originalnicodr** [comment]
Hey, thanks a lot for the detailed report and testing! As far as I understand this might be because of me doing stuff that doesn't work in the game because lack of support from UniverseLib (and it isn't done automatically by UnityExplorer), although I could be wrong. I have already encountered something like this before and reported it in #55.

I believe the only way I can "fix" this is by disabling panels and features that get executed automatically after the `Creating UI...` is printed and see which one is causing this. It would sadly mean that I will need to disable a feature or something dynamically to make it work, but maybe that feature isn't that important for what you want to do, so it might be useful in the end.

I don't own the game but will try to see if I can fix it on Hellpoint and send you a build of that for you to test it here. I don't know when that's gonna happen since I am quite busy at the moment, but I will make it my priority next time I get to work on this again.

Sorry for the inconvenience!

**Comment by Dteyn** [comment]
Awesome, that sounds good to me!

I forgot to mention, I did look through the issues already posted to see if this was already reported, but mostly skimmed over the titles and only clicked through a few. I missed the Hellpoint one - good to see this is already documented as a known issue.

Once you get the chance to work on it I'll be happy to test the fix to see if it works on my end. I'm fairly technically minded - although not very well versed in C# and very much a Unity newbie - but I'm happy to help test anything.

Thanks again for your help! :)

PS: Also, I renamed the issue to `[Bug]: Crash while Creating UI` since it doesn't seem to be related to `IL2CPP method has no corresponding pointer, aborting patch of UnityEngine.AssetBundle.UnloadAllAssetBundles`.

**Comment by Dteyn** [comment]
Good news! I've been able to narrow it down to the "Misc" UI panel.

I downloaded the source and set up for building it with VS, then tracked down the 'Creating UI...' line in `ExplorerCore.cs` which lead me to `UIManager.cs` where the various UI elements are created

I then added logging statements there, like this:

```c#
            UIPanels.Add(Panels.AutoCompleter, new AutoCompleteModal(UiBase));
            ExplorerCore.Log("- AutoCompleter panel added");
            UIPanels.Add(Panels.ObjectExplorer, new ObjectExplorerPanel(UiBase));
            ExplorerCore.Log("- ObjectExplorer panel added");
            ... etc
```

Then, when starting CUE, I could see which panel it hung up on:

```
[22:38:10.561] [CinematicUnityExplorer] Creating UI...
[22:38:10.569] [CinematicUnityExplorer] - Init complete
[22:38:10.604] [CinematicUnityExplorer] - TopNavBar created
[22:38:10.624] [CinematicUnityExplorer] - AutoCompleter panel added
[22:38:10.692] [CinematicUnityExplorer] - ObjectExplorer panel added
[22:38:10.715] [CinematicUnityExplorer] - Inspector panel added
[22:38:10.732] [CinematicUnityExplorer] - CSConsole panel added
[22:38:10.764] [CinematicUnityExplorer] - HookManager panel added
[22:38:10.772] [CinematicUnityExplorer] - Clipboard panel added
[22:38:10.800] [CinematicUnityExplorer] - ConsoleLog panel added
[22:38:10.821] [CinematicUnityExplorer] - Freecam panel added
[22:38:10.830] [CinematicUnityExplorer] - LightsManager panel added
[22:38:10.864] [CinematicUnityExplorer] - CamPaths panel added
[22:38:10.868] [CinematicUnityExplorer] - PostProcessingPanel added
[22:38:10.876] [CinematicUnityExplorer] - AnimatorPanel added
```

From this, I could tell it was the 'Misc' panel that was causing the crash. I then commented out the 'Misc' panel loading, and as a result CUE is now loading and running for me.

```
[22:40:37.563] [CinematicUnityExplorer] Creating UI...
[22:40:37.570] [CinematicUnityExplorer] - Init complete
[22:40:37.621] [CinematicUnityExplorer] - TopNavBar created
[22:40:37.641] [CinematicUnityExplorer] - AutoCompleter panel added
[22:40:37.710] [CinematicUnityExplorer] - ObjectExplorer panel added
[22:40:37.718] [CinematicUnityExplorer] - Inspector panel added
[22:40:37.751] [CinematicUnityExplorer] - CSConsole panel added
[22:40:37.783] [CinematicUnityExplorer] - HookManager panel added
[22:40:37.790] [CinematicUnityExplorer] - Clipboard panel added
[22:40:37.802] [CinematicUnityExplorer] - ConsoleLog panel added
[22:40:37.840] [CinematicUnityExplorer] - Freecam panel added
[22:40:37.848] [CinematicUnityExplorer] - LightsManager panel added
[22:40:37.882] [CinematicUnityExplorer] - CamPaths panel added
[22:40:37.886] [CinematicUnityExplorer] - PostProcessingPanel added
[22:40:37.894] [CinematicUnityExplorer] - AnimatorPanel added
[22:40:37.895] [CinematicUnityExplorer] - Misc panel SKIPPED
[22:40:37.904] [CinematicUnityExplorer] - Options panel added
[22:40:37.908] [CinematicUnityExplorer] - UIInspectorResults panel added
[22:40:37.912] [CinematicUnityExplorer] - MouseInspector added
[22:40:38.510] [CinematicUnityExplorer] - Init methods completed
[22:40:38.512] [CinematicUnityExplorer] - Failsafe done
[22:40:38.512] [CinematicUnityExplorer] - Done initializing
[22:40:38.513] [CinematicUnityExplorer] - COMPLETE
[22:40:38.513] [CinematicUnityExplorer] CinematicUnityExplorer 1.0.0 (IL2CPP) initialized.
```

After this, there does seem to be some errors popping up:

```
[22:40:38.569] [CinematicUnityExplorer] [ERROR] Exception in IL2CPP-to-Managed trampoline, not passing it to il2cpp: System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <e1319b7195c343e79b385cd3aa43f5dc>:0 

  at UnityExplorer.UI.UIManager.GetPanel[T] (UnityExplorer.UI.UIManager+Panels panel) [0x00000] in <41a310062afd405c91bff14365369972>:0 

  at UnityExplorer.KeypressListener.maybeTakeScreenshot () [0x00001] in <41a310062afd405c91bff14365369972>:0 

  at UnityExplorer.KeypressListener.Update () [0x0000f] in <41a310062afd405c91bff14365369972>:0 

  at (wrapper dynamic-method) UnityExplorer.KeypressListener.Trampoline_VoidThisUnityExplorer.KeypressListenerUpdate(intptr,UnhollowerBaseLib.Runtime.Il2CppMethodInfo*)
[22:40:38.648] [CinematicUnityExplorer] [ERROR] Exception in IL2CPP-to-Managed trampoline, not passing it to il2cpp: System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <e1319b7195c343e79b385cd3aa43f5dc>:0 

  at UnityExplorer.UI.UIManager.GetPanel[T] (UnityExplorer.UI.UIManager+Panels panel) [0x00000] in <41a310062afd405c91bff14365369972>:0 

  at UnityExplorer.KeypressListener.maybeTakeScreenshot () [0x00001] in <41a310062afd405c91bff14365369972>:0 

  at UnityExplorer.KeypressListener.Update () [0x0000f] in <41a310062afd405c91bff14365369972>:0 

  at (wrapper dynamic-method) UnityExplorer.KeypressListener.Trampoline_VoidThisUnityExplorer.KeypressListenerUpdate(intptr,UnhollowerBaseLib.Runtime.Il2CppMethodInfo*)
```

This happens for quite a lot of lines, before it settles down and gets to this:

```
[22:46:09.755] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.Vignette
[22:46:09.759] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.Bloom
[22:46:09.764] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.Grain
[22:46:09.768] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.DepthOfField
[22:46:09.772] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.LensDistortion
[22:46:09.776] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.ChromaticAberration
[22:46:09.780] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.AmbientOcclusion
[22:46:09.785] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.AutoExposure
[22:46:09.789] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.ScreenSpaceReflections
[22:46:09.793] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Vignette
[22:46:09.797] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Bloom
[22:46:09.801] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Grain
[22:46:09.806] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Fog
[22:46:09.810] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.DepthOfField
[22:46:09.814] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Tonemapping
[22:46:09.819] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.LensDistortion
[22:46:09.823] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.ChromaticAberration
[22:46:09.827] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.AmbientOcclusion
[22:46:09.831] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.AutoExposure
[22:46:09.836] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.ScreenSpaceReflections
[22:46:09.840] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Vignette
[22:46:09.844] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Bloom
[22:46:09.848] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Grain
[22:46:09.852] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Fog
[22:46:09.856] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.DepthOfField
[22:46:09.861] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Tonemapping
[22:46:09.865] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.LensDistortion
[22:46:09.869] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.ChromaticAberration
[22:46:09.873] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.AmbientOcclusion
[22:46:09.877] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.AutoExposure
[22:46:09.882] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.ScreenSpaceReflections
[22:48:40.772] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.Vignette
[22:48:40.774] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.Bloom
[22:48:40.775] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.Grain
[22:48:40.777] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.DepthOfField
[22:48:40.779] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.LensDistortion
[22:48:40.780] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.ChromaticAberration
[22:48:40.782] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.AmbientOcclusion
[22:48:40.783] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.AutoExposure
[22:48:40.785] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.PostProcessing.ScreenSpaceReflections
[22:48:40.786] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Vignette
[22:48:40.788] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Bloom
[22:48:40.790] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Grain
[22:48:40.791] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Fog
[22:48:40.793] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.DepthOfField
[22:48:40.794] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.Tonemapping
[22:48:40.796] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.LensDistortion
[22:48:40.798] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.ChromaticAberration
[22:48:40.800] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.AmbientOcclusion
[22:48:40.801] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.AutoExposure
[22:48:40.803] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.HighDefinition.ScreenSpaceReflections
[22:48:40.804] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Vignette
[22:48:40.806] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Bloom
[22:48:40.807] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Grain
[22:48:40.809] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Fog
[22:48:40.810] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.DepthOfField
[22:48:40.813] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.Tonemapping
[22:48:40.814] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.LensDistortion
[22:48:40.816] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.ChromaticAberration
[22:48:40.817] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.AmbientOcclusion
[22:48:40.819] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.AutoExposure
[22:48:40.821] [CinematicUnityExplorer] Couldn't find UnityEngine.Rendering.Volume.ScreenSpaceReflections
```

My guess is that the errors were something introduced by commenting out the Misc panel loading. It's also possible that my build didn't go correctly - the CUE .dll file seemed to come in a lot smaller at 405kb, and the Release folder also had `Iced.dll`, `mcs.dll` and `Tomlet.dll` which I had to move to the ML Mods folder. Because of this I'm sure I did something wrong with the build process (I'm very new to VS / C# / UE) - but despite that, it did seem to load correctly.

However, despite these errors, CUE did seem to function fine during my testing. I ran it for about an hour doing usual UE stuff like object inspection etc and all ran perfectly fine. I also tried adding a new light source and that worked as expected also.

So, it looks like the issue lies somewhere in the Misc panel init. If there's anything else I can do to assist with this, I'm happy to follow up on anything here. Just let me know!

Thanks again for all your work on this - I'm super stoked to use it and make some awesome captures! 😃

**Comment by originalnicodr** [comment]
I was actually gonna suggest to do exactly that! Thanks a ton for the testing!

The key listener errors are indeed due to the Misc panel not initializing, but they aren't really important if the rest of the mod is working as intended.

The list of "couldn't find" prints is normal when looking for postprocessing effects, but I now see how it can get confusing. Will comment it on the next release to have more straightforward logs.

The smaller .dll size thing seems to be because there was a problem in the build process and it couldn't link all the .DLLs into a single one. Can't really say what happened without the console output when building the dll, but don't worry I will take care of this when I work on fixing the problem. Glad you managed to work around that.

Thanks once again for narrow down where the problem lies! Will do my best to get on this as soon as I can.

And please, hit me up when you upload something using the mod, would love to check it out!

**Comment by originalnicodr** [comment]
I cannot assure you that the random crashes will be fixed, but maybe your logs can help us out. And no worries, .net isn't my forte either hahaha.

I am sadly hands full with finishing my final thesis atm, but I might be able to check it out in the following weeks. Will let you know if I have a new version you can test.

I have actually thought about serializing camera paths (see #17). This would have been my next thing to work on after controller support (#20), but I worked on the project for months and I need to at least release a first version :stuck_out_tongue: 

Regarding editing coordinates of path nodes, you can actually do that already! Just move the camera where you want your new node position and rotation to be and click on the first button on the node row on the Cam Paths panel. The one that says "Copy Camera pos and ro" (I am sorry, I suck at naming things rofl). If you also need to copy the FoV over just press on the button next to it, "Copy Camera FoV".

For further fine-tuning, you can add nodes in the middle of the list by simply adding a new node, which would be added at the end of the list, and then just move it upwards one step at a time by pressing the upwards arrow button until it reaches the part of the "timeline" you want it to be. I suggest turning on the "Visualize camera paths" options when doing this for easier setup.

**Comment by originalnicodr** [comment]
Hey! Managed to fix Hellpoint problem on 66773b300fac50ca26d22851ce82d11198b4a95d, and since it was similar to what was happening here (even has a similar Unity version, `2021.1.27f1`) it maybe fixed this error as well. Could you give it a try?

[Here are the nightly builds](https://github.com/originalnicodr/CinematicUnityExplorer/actions/runs/8963257548).

**Comment by Dteyn** [comment]
> Hey! Managed to fix Hellpoint problem on [66773b3](https://github.com/originalnicodr/CinematicUnityExplorer/commit/66773b300fac50ca26d22851ce82d11198b4a95d), and since it was similar to what was happening here (even has a similar Unity version, `2021.1.27f1`) it maybe fixed this error as well. Could you give it a try?
> 
> [Here are the nightly builds](https://github.com/originalnicodr/CinematicUnityExplorer/actions/runs/8963257548).

Hey! Sorry for not getting back to you sooner. I had some things come up in life so haven't been able to work on this project lately, but I do plan on getting back to it at some point. It's still in the back of my mind.

Also wanted to let you know that I did try the recent update since you fixed the Hellpoint problem, and can confirm it fixed the problem on my end as well. I only tested it briefly, but it did start up without any issues and the Misc panel was created. I expect all the problems I was facing are resolved. This issue can be closed now.

Also, thanks for your earlier reply on the Cam Paths. That's very helpful and I will keep that in mind once I return to this project. I have full intentions of seeing it through once I finish off some other projects I'm working on. Really looking forward to digging back into this.

Thanks again for your hard work on this project!
